get_modis.py

```diff
@@ -241,91 +265,237 @@ def get_modisfiles(username, password, platform, product, year, tile, proxy,
                                                      "%j/%Y")) for i in
              range(doy_start, doy_end)]
     url = "%s/%s/%s/" % (base_url, platform, product)
-    dates = parse_modis_dates(url, dates, product, out_dir, ruff=ruff)
-    them_urls = []
-    for date in dates:
-        r = requests.get("%s/%s" % (url, date), verify=False)
-        for line in r.text.split("\n"):
-            if line.decode().find(tile) >= 0:
-                if line.decode().find(
-                        ".hdf") >= 0 > line.decode().find(".hdf.xml"):
-                    fname = line.decode().split("href=")[1].split(">")[0].strip('"')
-
-                    if not os.path.exists(os.path.join(out_dir, fname)):
-                        them_urls.append("%s/%s/%s" % (url, date, fname))
-                    else:
-                        if verbose:
-                            LOG.info("File %s already present. Skipping" % fname)
-
-    with requests.Session() as s:
-        s.auth = (username, password)
-        for the_url in them_urls:
-            r1 = s.request('get', the_url)
-            r = s.get(r1.url, stream=True)
-            if not r.ok:
-                raise IOError("Can't start download... [%s]" % fname)
-            file_size = int(r.headers['content-length'])
-            fname = the_url.split("/")[-1]
-            LOG.info("Starting download on %s(%d bytes) ..." %
-                     (os.path.join(out_dir, fname), file_size))
-            with open(os.path.join(out_dir, fname), 'wb') as fp:
-                for chunk in r.iter_content(chunk_size=CHUNKS):
-                    if chunk:
-                        fp.write(chunk)
-                fp.flush()
-                os.fsync(fp)
+    dates_available = None
```

Comment:
I have another script that scans folders in parallel, gathers all the URLs and then downloads them in parallel (you're only allowed a couple of slots for download, but scanning is much faster). Maybe worth considering...

Reply:
thanks for the info - could be useful. At the moment I'm working primarily with MOD10A1. The product sizes are small enough that I don't really need to worry about parallelising download.